packages/components/src/FilterBar/subcomponents/FilterMS.tsx

```diff
@@ -0,0 +1,57 @@
+import React from "react"
+import { FilterMultiSelect, getSelectedOptionLabels } from "@kaizen/select"
+import { RootProps } from "@kaizen/select/src/FilterMultiSelect/components/Root"
+import { useFilterBarContext } from "../context/FilterBarContext"
+
+export type FilterBarMultiSelectProps = Omit<RootProps, "trigger" | "label"> & {
+  id: string
+}
+
+export const FilterBarMultiSelect = ({
+  items,
+  children,
+  ...props
+}: FilterBarMultiSelectProps): JSX.Element | null => {
+  const { getFilterState, updateSelectedValue, toggleOpenFilter, hideFilter } =
+    useFilterBarContext()
+
+  const filterState = getFilterState(props.id)
+
+  return (
+    <FilterMultiSelect
+      label={filterState.label}
+      // Convert the incoming FilterBar state to a Set (internal FilterMultiSelect state)
+      selectedKeys={new Set(filterState.selectedValue)}
+      onSelectionChange={(keys): void => {
+        // Convert the internal FilterMultiSelect state (Set) to an Array for FilterBar state
+        updateSelectedValue(props.id, Array.from(keys))
+      }}
+      items={items}
+      isOpen={filterState.isOpen ?? false}
+      onOpenChange={(open): void => toggleOpenFilter(props.id, open)}
+      trigger={(): JSX.Element =>
+        filterState.isRemovable ? (
+          <FilterMultiSelect.RemovableTrigger
+            selectedOptionLabels={getSelectedOptionLabels(
+              filterState.selectedValue,
+              items
+            )}
+            onRemove={() => hideFilter(props.id)}
+            label={filterState.label}
+          />
+        ) : (
+          <FilterMultiSelect.TriggerButton
+            selectedOptionLabels={getSelectedOptionLabels(
+              filterState.selectedValue,
+              items
+            )}
+            label={filterState.label}
+          />
+        )
```

Comment:
This component isn't styled correctly. Is it possible to retrofit it to FilterButton?

Reply:
As discussed, we are descoping retrofitting to use `Filter` Abstraction due to significant rework required to `FilterMultiSelect` and will add this to the roadmap.

To summarise:
- A new `MultiSelect` is on the horizon which can be built with the `Filter` abstraction in mind so we can potentially port the `FilterMultiSelect` to use it.
- We ant to maintain high compatibility with existing implementations of `FilterMultiSelect`, any potential rework of the API will disrupt adoption.